tests/foreman/cli/test_variables.py

```diff
@@ -0,0 +1,1108 @@
+
+        @CaseLevel: Integration
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_list_variables_by_puppetclass_name(self):
+        """List all the variables for specific puppet class by name.
+
+        @id: 5df71b8d-ab85-43cd-83f2-a0afe9e467a0
+
+        @steps:
+
+        1. List all the variables for specific puppet class by its name.
+
+        @assert: Variables listed for specific Puppet class by name.
+
+        @CaseLevel: Integration
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_list_variables_by_puppetclass_id(self):
+        """List all the variables for specific puppet class by id.
+
+        @id: ba261c80-699c-4b75-9e07-b80cb0a80c55
+
+        @steps:
+
+        1. List all the variables for specific puppet class by its id.
+
+        @assert: Variables listed for specific Puppet class by id.
+
+        @CaseLevel: Integration
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_create(self):
+        """Create a Smart Variable.
+
+        @id: 4a8589bf-7b11-48e8-a25d-984bea2ba676
+
+        @steps:
+
+        1. Create a smart Variable with Valid name and default value.
+
+        @assert: The smart Variable is created successfully.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_negative_create(self):
+        """Smart Variable is not created with invalid data.
+
+        @id: 09a67cb6-5de0-41b3-90f4-593323936c6c
+
+        @steps:
+
+        1. Create a smart Variable with Invalid name and valid default value.
+
+        @assert: The smart Variable is not created.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_delete_smart_variable_by_id(self):
+        """Delete a Smart Variable by id.
+
+        @id: 19fedbdf-48a1-46a7-b184-615a0efd7b4e
+
+        @steps:
+
+        1. Delete a smart Variable by id.
+
+        @assert: The smart Variable is deleted successfully.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_delete_smart_variable_by_name(self):
+        """Delete a Smart Variable by name.
+
+        @id: 19fedbdf-48a1-46a7-b184-615a0efd7b4e
+
+        @steps:
+
+        1. Delete a smart Variable by name.
+
+        @assert: The smart Variable is deleted successfully.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_update_variable_puppet_class(self):
+        """Update Smart Variable's puppet class.
+
+        @id: 6c3e2da9-420c-4e39-8b71-e5be6b605bd7
+
+        @steps:
+
+        1. Create a smart variable with valid name and default value.
+        2. Add this variable to some host/hostgroup.
+        2. Update the variable with some else puppet class than existing.
```

Comment:
I think these steps needs refinement. 
a) Please update the step numbers. If I understand correctly: Third step should be like: `Update the puppet-class associated to the variable created in step1`